## Comparing files

In [2]:
import pandas as pd

In [3]:
df = pd.read_pickle('../data/data_from_remove_from_dataset.pkl')
df.shape

(516406, 44)

In [4]:
df1 = pd.read_pickle('../data/data_w_transformed_census_and_removed_invalid_rows_and_cols_and_fixed_zips_and_descs_and_dropped_latlongs_and_more_dropped_cols_and_rows_and_actual_descs.pkl')
df1.shape

(520793, 29)

In [45]:
new_add = pd.read_pickle('../data/data_from_add_to_dataset.pkl')
new_add.shape

(905335, 66)

In [5]:
new = df
old = df1

In [16]:
520793 - 516406

4387

## Cols in new not in old, vice versa

In [6]:
set(new.columns).difference(old.columns)

{'ClosedPhoto',
 'Geocoded_Location',
 'LOCATION_ZIPCODE',
 'bedroom_std_dev',
 u'case_enquiry_id',
 'earned_income_per_capita',
 'housing_std_dev',
 'income_std_dev',
 'queue_wk',
 'rent_std_dev',
 'school_std_dev',
 u'specific_location',
 u'title',
 'tract_and_block_group',
 'value_std_dev'}

In [7]:
set(old.columns).difference(new.columns)

set()

So that's fine.

## rows in old not in new, vice versa

In [17]:
len(set(new.CASE_ENQUIRY_ID).difference(old.CASE_ENQUIRY_ID))

1766

In [43]:
len(set(old.CASE_ENQUIRY_ID).difference(new.CASE_ENQUIRY_ID))

6153

In [47]:
len(set(old.CASE_ENQUIRY_ID).difference(new_add.CASE_ENQUIRY_ID))

0

When I removed rows, I removed an extra 6153 that I shouldn't have.

In addition, I should have removed an extra 1766.

## which are the ones i shld've removed?

In [55]:
new[~new.CASE_ENQUIRY_ID.isin(old.CASE_ENQUIRY_ID)].head(2)[['LATITUDE', 'neighborhood_from_zip']]

LATITUDE neighborhood_from_zip
905389   42.3594         Jamaica Plain
905296   42.3594         Jamaica Plain

I think it's sth related to the defauly ones at City Hall.

## inspecting

In [19]:
new.head(2).loc[:, 'LONGITUDE':]

LONGITUDE                Source    Geocoded_Location  case_enquiry_id  \
905425   -71.0680  Citizens Connect App   (42.2809, -71.068)              NaN   
905235   -71.0067  Citizens Connect App  (42.3871, -71.0067)              NaN   

       description specific_location title  COMPLETION_TIME  \
905425         NaN               NaN   NaN         0.918333   
905235         NaN               NaN   NaN         2.325000   

       tract_and_block_group  queue_wk       ...        housing  \
905425               1004002       NaN       ...            own   
905235               0511011       NaN       ...           rent   

        housing_std_dev  bedroom  bedroom_std_dev     value  value_std_dev  \
905425        26.870058        3        61.512329  350000.0      20.979404   
905235       153.442172        2        90.187397  225000.0      14.876362   

        rent  rent_std_dev  income  income_std_dev  
905425  1750     19.162161  112500       28.616720  
905235  1125     35.203652  112500       30.615832  

[2 rows x 35 columns]

In [23]:
new[['CASE_ENQUIRY_ID', 'TYPE', 'queue_wk', 'case_enquiry_id', 'description', 'housing']].isnull().sum()

CASE_ENQUIRY_ID         0
TYPE                    0
queue_wk           328376
case_enquiry_id    328376
description        353430
housing                 0
dtype: int64

## >>>

In [24]:
df_descs = pd.read_pickle('../data/data_w_descs.pkl')

In [28]:
df_descs.head()[['CASE_ENQUIRY_ID', 'case_enquiry_id', 'description']]

CASE_ENQUIRY_ID  case_enquiry_id  \
0     101000958209     1.010010e+11   
1     101001243239     1.010012e+11   
2     101000945083              NaN   
3     101000768355              NaN   
4     101001619220     1.010016e+11   

                                         description  
0  Public safety sign: No | Type of sign: no stop...  
1                    Electric or gas light: Electric  
2                                                NaN  
3                                                NaN  
4  Additional information: Approx 4 to 6 flags/ e...

In [29]:
df_descs.shape

(905650, 37)

In [32]:
df.head(1).loc[:, 'LONGITUDE':]

LONGITUDE                Source   Geocoded_Location  case_enquiry_id  \
905425    -71.068  Citizens Connect App  (42.2809, -71.068)              NaN   

       description specific_location title  COMPLETION_TIME  \
905425         NaN               NaN   NaN         0.918333   

       tract_and_block_group  queue_wk       ...        housing  \
905425               1004002       NaN       ...            own   

        housing_std_dev  bedroom  bedroom_std_dev     value  value_std_dev  \
905425        26.870058        3        61.512329  350000.0      20.979404   

        rent  rent_std_dev  income  income_std_dev  
905425  1750     19.162161  112500        28.61672  

[1 rows x 35 columns]

In [34]:
df.loc[:, 'case_enquiry_id': 'title'].isnull().sum()

case_enquiry_id      328376
description          353430
specific_location    506210
title                334494
dtype: int64

In [40]:
df[(df.CASE_ENQUIRY_ID != df.case_enquiry_id) & df.description.notnull()].shape[0]

0

In [38]:
df.specific_location[df.specific_location.notnull()].head()

905500    \n\n\n\n\n\n\n\nTimestamp\nDescription\n\n\n\n...
905537    \n\n\n\n\n\n\n\nTimestamp\nDescription\n\n\n\n...
904806               \n Location: \n\t\t\t\tRoadway\n\t\t\t
904428              \n Location: \n\t\t\t\tSidewalk\n\t\t\t
903289    \n\nNeedle Quantity: One | Property Location T...
Name: specific_location, dtype: object